volatility3/framework/plugins/windows/amcache.py

```diff
@@ -0,0 +1,603 @@
+
+    if expected_type == reg_extensions.RegValueTypes.REG_SZ:
+        if isinstance(value, bytes):
+            return value.decode("utf-16le", errors="replace").rstrip("\u0000")
+        else:
+            return renderers.UnparsableValue()
+
+    if expected_type in [
+        reg_extensions.RegValueTypes.REG_DWORD,
+        reg_extensions.RegValueTypes.REG_QWORD,
+        reg_extensions.RegValueTypes.REG_DWORD_BIG_ENDIAN,
+    ]:
+        if isinstance(value, int):
+            return value
+        else:
+            return renderers.UnparsableValue()
+
+    return value
+
+
+def _sort_entries(entry_tuple):
+    program_id, _ = entry_tuple
+    key = program_id if isinstance(program_id, str) else ""
+    return key
+
+
+class Amcache(interfaces.plugins.PluginInterface, timeliner.TimeLinerInterface):
+    """Scans for windows services."""
+
+    _required_framework_version = (2, 0, 0)
+    _version = (1, 0, 0)
+
+    def generate_timeline(
+        self,
+    ) -> Iterator[Tuple[str, timeliner.TimeLinerType, datetime.datetime]]:
+        for _, (
+            path,
+            _,
+            last_mod,
+            last_mod_2,
+            install_time,
+            compile_time,
+            _,
+            _,
+            _,
+            entry_type,
+        ) in self._generator():
+            if isinstance(last_mod, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} registry key modified", timeliner.TimeLinerType.MODIFIED, last_mod
+            if isinstance(last_mod_2, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} STANDARD_INFORMATION create time", timeliner.TimeLinerType.CREATED, last_mod_2
+            if isinstance(install_time, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} installed", timeliner.TimeLinerType.CREATED, install_time
+            if isinstance(compile_time, datetime.datetime):
+                yield f"Amcache: {entry_type} {path} compiled (PE metadata)", timeliner.TimeLinerType.MODIFIED, compile_time
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        # Since we're calling the plugin, make sure we have the plugin's requirements
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Windows kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="hivelist", plugin=hivelist.HiveList, version=(1, 0, 0)
+            ),
+        ]
+
+    @classmethod
+    def get_amcache_hive(
+        cls,
+        context: interfaces.context.ContextInterface,
+        config_path: str,
+        kernel: interfaces.context.ModuleInterface,
+    ) -> Optional[registry.RegistryHive]:
+        """Retrieves the `Amcache.hve` registry hive from the kernel module, if it can be located."""
+        return next(
```

Comment:
Again with the next.  Happy you're consistent but it still makes me look around for a list that doesn't seem to exist...  5:P